### PR TITLE
additions to puppet completion

### PR DESCRIPTION
**Pull request by sluetze**
Hi,
since I was missing some subcommands in the completion I added them. Since I'm a bash-completion virgin, please review closely.

Tested on CentOS 7 using bash 4.2.46 and puppet 5.0

### CODE REVIEW

File: completions/puppet [hunk 1]
```diff
@@ -276,6 +281,27 @@ _puppet()
...
+		--facts)
+		    _filedir json
+		    return
+		    ;;
+		--node)
+		    _known_hosts_real -- "$cur"
+		    return
+		    ;;
+		--merge)
+		    _puppet_hash_opts
```
Comment (scop) [inline completions/puppet]: Please just inline _puppet_hash_opts here if it's not going to be used from anywhere else.

File: completions/puppet [hunk 2]
```diff
@@ -101,16 +101,16 @@ _puppet()
             ;;
         puppet)
             case ${words[1]} in
-                agent|apply|cert|describe|doc|filebucket|kick|lookup|master|parser|queue|resource)
+                agent|apply|catalog|cert|describe|doc|epp|filebucket|help|kick|lookup|man|master|module|node|parser|report|resource)
                     subcommand=${words[1]}
```
Comment (scop) [inline completions/puppet]: This seems to add more than just epp, please adjust the commit message accordingly.

Also, I suggest splitting the commit to two, one for adding commands and another removing them.

Comment (scop) [inline completions/puppet]: Well, perhaps all these can be squashed to one commit after all (I can take a look at it on merge), so please disregard the split comment above.

**Comment by sluetze** [comment]
hi - did I miss anything in the changes?

File: completions/puppet [hunk 3]
```diff
@@ -96,16 +96,16 @@ _puppet()
             ;;
         puppet)
             case ${words[1]} in
-                agent|apply|cert|describe|doc|filebucket|kick|master|parser|queue|resource)
+                agent|apply|catalog|cert|describe|doc|epp|filebucket|help|kick|lookup|man|master|module|node|parser|report|resource)
```
Comment (scop) [inline completions/puppet]: Please split long lines, see existing completions for examples.

File: completions/puppet [hunk 4]
```diff
@@ -168,7 +168,26 @@ _puppet()
                     return
             esac
             ;;
-        cert)
+        catalog)
+        case $prev in
```
Comment (scop) [inline completions/puppet]: Indentation for the added catalog) block contents seems to be off, please follow style of existing blocks.

File: completions/puppet [hunk 5]
```diff
@@ -168,7 +168,26 @@ _puppet()
...
+        *)
```
Comment (scop) [inline completions/puppet]: Redundant case statement, only *) there?

File: completions/puppet [hunk 6]
```diff
@@ -234,6 +253,12 @@ _puppet()
                     return
             esac
             ;;
+        epp)
+            case $prev in
+                *)
```
Comment (scop) [inline completions/puppet]: Redundant case statement, only *) there?

File: completions/puppet [hunk 7]
```diff
@@ -289,6 +333,28 @@ _puppet()
                     return
             esac
             ;;
+        module)
+        case $prev in
+        *)
```
Comment (scop) [inline completions/puppet]: Redundant case statement, only *) there?

File: completions/puppet [hunk 8]
```diff
@@ -289,6 +333,28 @@ _puppet()
                     return
             esac
             ;;
+        module)
+        case $prev in
```
Comment (scop) [inline completions/puppet]: Indentation of module) block contents seems off, please follow style of existing blocks.

File: completions/puppet [hunk 9]
```diff
@@ -289,6 +333,28 @@ _puppet()
...
+            _puppet_subcmd_opts "$1" $subcommand
+            else
+            COMPREPLY=( $( compgen -W 'build changes generate install list search uninstall upgrade'  -- "$cur" ) )
+            fi
+            return
+        esac
+        ;;
+    node)
+        case $prev in
+        *)
```
Comment (scop) [inline completions/puppet]: Redundant case statement, only *) there?

File: completions/puppet [hunk 10]
```diff
@@ -289,6 +333,28 @@ _puppet()
...
+                    if [[ "$cur" == -* ]]; then
```
Comment (scop) [inline completions/puppet]: Indentation one level too deep here.

File: completions/puppet [hunk 11]
```diff
@@ -301,22 +367,16 @@ _puppet()
...
-                    if [[ "$cur" == -* ]]; then
-                        _puppet_subcmd_opts "$1" $subcommand
-                    else
-                        _filedir
-                    fi
-                    return
-            esac
-            ;;
-        resource|*)
+    report)
```
Comment (scop) [inline completions/puppet]: Indentation of this block seems to be off (too shallow altogether).

**Pull request closed without merging**